wcfsetup/install/files/lib/data/user/online/UsersOnlineList.class.php

```diff
@@ -157,19 +158,23 @@ public function checkRecord() {
 	 */
 	public static function isVisible($userID, $canViewOnlineStatus) {
 		if (WCF::getSession()->getPermission('admin.user.canViewInvisible') || $userID == WCF::getUser()->userID) return true;
+
+		$data = ['result' => false, 'userID' => $userID, 'canViewOnlineStatus' => $canViewOnlineStatus];
 		
 		switch ($canViewOnlineStatus) {
 			case 0: // everyone
-				return true;
+				$data['result'] = true;
```

Comment:
One last thing: This is now a fall-through which serves no purposes and also has no side-effect. Nevertheless, this looks really odd and I was scratching my head while reading your code.

I failed to mark this in my previous review request, but I want to merge this whole change in a single squash commit without having a loosely coupled commit from me to fix this. Please do me a favor and add a `statement` and this is ready for merge.